src/VerifyingPaymaster.sol

```diff
@@ -320,21 +315,24 @@ contract VerifyingPaymaster is BasePaymaster, Ownable2Step {
 
         // Reject if should restrict bundlers and bundler not on allowlist to prevent siphoning of funds
         if (!c.allowAnyBundler && !isBundlerAllowed[tx.origin]) {
-            revert BundlerNotAllowed();
+            revert BundlerNotAllowed(tx.origin);
         }
 
         // Attempt transfer if token is set and not mode not postOpReverted
         if (c.token != address(0) && mode != PostOpMode.postOpReverted) {
             // get current gas price and token cost
             uint256 gasPrice = _min(c.maxFeePerGas, c.maxPriorityFeePerGas + block.basefee);
-            uint256 actualTokenCost = _calculateTokenCost(actualGasCost + c.postOpGasCost * gasPrice, c.exchangeRate);
+            uint256 actualTokenCost = _calculateTokenCost(actualGasCost + c.postOpGas * gasPrice, c.exchangeRate);
 
             // If not prepaid transfer full amount to receiver else refund sender difference and transfer to receiver
             if (c.prepaidAmount == 0) {
-                IERC20(c.token).safeTransferFrom(c.sender, c.receiver, actualTokenCost);
+                ERC20(c.token).safeTransferFrom(c.sender, c.receiver, actualTokenCost);
             } else {
-                IERC20(c.token).safeTransfer(c.sender, c.prepaidAmount - actualTokenCost);
-                IERC20(c.token).safeTransfer(c.receiver, actualTokenCost);
+                uint256 refund = c.prepaidAmount - actualTokenCost;
+                if (refund > 0) {
+                    ERC20(c.token).safeTransfer(c.receiver, refund);
+                }
```

Comment:
I think the refund was accidentally changed to send to `c.receiver` instead of `c.sender`, I'd suggest the following quick fix:

```diff
if (refund > 0) {
-   ERC20(c.token).safeTransfer(c.receiver, refund);
+   ERC20(c.token).safeTransfer(c.sender, refund);
}
```

Reply:
Yup! Fixed thanks